src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -418,9 +436,17 @@ public void Reverse()
         public void Reverse(int index, int count)
         {
             if (index < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(index));
+            }
             if (count < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(count));
+            }
+            if (Count - index < count)
+            {
+                throw new ArgumentException(SR.ArgumentInvalidOffLen);
+            }
```

Comment:
No. Array Length can be above then Builder.Count, so array does not throw any exception and revers elements that out of bounds.